src/main/java/frc/robot/subsystems/endEffector/TOFSensor.java

```diff
@@ -0,0 +1,82 @@
+package frc.robot.subsystems.endEffector;
+
+import com.playingwithfusion.TimeOfFlight;
+import com.playingwithfusion.TimeOfFlight.RangingMode;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class TOFSensor {
+  public double m_distance_EMA;
+  public double m_dist_SDEV_sq;
+  public String mode = "Short";
+
+  int m_rangeX0;
+  int m_rangeY0;
+  int m_rangeX1;
+  int m_rangeY1;
+
+  private TimeOfFlight sensor;
+
+  public TOFSensor(int CAN_ID) {
+    sensor = new TimeOfFlight(CAN_ID);
+    RoI(0, 0, 15, 15);
+    sensor.setRangingMode(RangingMode.Short, 0.24); // Will need to test. must be between 24-1000ms
+    m_distance_EMA = sensor.getRange();
+    m_dist_SDEV_sq = 0;
+
+  }
+
+  public TimeOfFlight.Status getStatus() {
+    return sensor.getStatus();
+  }
+  
+  public void RoI(int rangeX0, int rangeY0, int rangeX1, int rangeY1) {
```

Comment:
```suggestion
  public void setRangeOfInterest(int rangeX0, int rangeY0, int rangeX1, int rangeY1) {
```

Honestly this is me being nickpicky but `RoI` as a method name is somewhat confusing. I'd suggest using `setRangeOfInterest` (just like PWF does it) instead. The current naming style almost got me confused thinking that `RoI` was a constructor, not a method.